[master] stalekey engine fixes

garethgreenaway opened the pull request:
### What does this PR do?

Fixing stalekey engine so it deletes the keys when they are a list.

Breaking out various functionality so it's easier to test.

Writing tests.

### What issues does this PR fix or reference?
#55057 

### Tests written?
**[NOTICE] Bug fixes or features added to Salt require tests.**
Please review the [test documentation](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite.
Yes
### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

garethgreenaway committed:
Initial work to allow running multiple instances of a Salt engine.

garethgreenaway committed:
Adding some documention and some information to the release notes about new engine_type parameter

garethgreenaway committed:
Adding a test to test engine_type parameter.

garethgreenaway committed:
Moving unit.engines.test_engines to unit.test_engines.

garethgreenaway committed:
Moving unit.engines.test_engines to unit.test_engines.

garethgreenaway committed:
Renaming engine_type to engine_module for configuration parameter.

mbunkus committed:
File module: make file.remove with other types (FIFOs, sockets)

The current checks only take regular files, symlinks and directories
into account. For all other types (FIFOs, sockets) the module doesn't
even attempt to remove them.

The new code only special-cases directories. For all other types of
existing file system entries a simple os.remove() is attempted.

This is the first part of fixing #35621.

mbunkus committed:
File state: make symlink deal with other existing types (FIFOs, sockets)

The current checks only take regular files, symlinks and directories
into account. For all other types (FIFOs, sockets) the state doesn't
even attempt to remove them; it simply returns an error that something
exists.

The new code keeps handling of symlinks as-is while generalizing all
other cases by only checking if they exist, and if they do, letting
the 'file.remove' module function handle the removal if 'force' is
true.

The 'file.remove' module function suffered from the same problem. It
was fixed to handle all types in the previous commit.

This is the second part of the fix for #35621.

mbunkus committed:
update Neon release notes about file.symlink state/file.remove module

twangboy committed:
Check for key before trying to list it

twangboy committed:
Add tests, improve error handling

twangboy committed:
Fix some lint

garethgreenaway committed:
fixing stalekey engine so it deletes the keys when they are a list.

garethgreenaway committed:
Adding additional test.  Some lint cleanup.

s0undt3ch commented on salt/engines/stalekey.py:
```diff
@@ -39,41 +39,103 @@
+
+
 def start(interval=3600, expire=604800):
+    '''
```
I'm going to pull a @dwoz on you :)
For reference, https://github.com/saltstack/salt/pull/55047#discussion_r336696340

Let's make start accept `_check_running_callable=None`

Then the `while` code Bellow becomes:
```python
while True:
    # do stuff
    If callable(_check_running_callable) and _check_running_callable() is False:
        break
```

No?

garethgreenaway replied:
Not against the idea, just curious, what is the advantage?

garethgreenaway replied:
Actually the `_running` function was being used when I was attempting to run `start` from tests, which didn't work out.  So I can remove that.

s0undt3ch replied:
Not having a function for us just to patch during tests, ie, dependency injection.

s0undt3ch reviewed (changes requested):
You need to pull master and rebase too.

garethgreenaway committed:
Removing the _running and the logic in start that was calling it.  Not needed anymore.

garethgreenaway committed:
Lint fix, removing unnecessary code.

arykalin committed:
Copy pasting changes from branch venafi-fix

arykalin committed:
Update code to latest vcert

arykalin committed:
Add `vcert` requirement

s0undt3ch committed:
Update to latest `unittest-xml-reporting`. Fix missing `.fileno` method.

s0undt3ch committed:
Update to latest `unittest-xml-reporting`. Fix missing `.fileno` method.

s0undt3ch committed:
Use the right base class. Mark tests which require root.

waynew commented on tests/unit/engines/test_stalekey.py:
```diff
@@ -0,0 +1,93 @@
+from tests.support.unit import skipIf, TestCase
+from tests.support.mock import (
+    NO_MOCK,
+    NO_MOCK_REASON,
```
Gonna need to remove `NO_MOCK` and `NO_MOCK_REASON` here.

waynew commented on tests/unit/engines/test_stalekey.py:
```diff
@@ -0,0 +1,93 @@
+        pass
+
+
+@skipIf(NO_MOCK, NO_MOCK_REASON)
```
This no mock skip should also be removed

cbosdo committed:
Add virt.network_get_xml function

Users may want to see the full XML definition of a network.

cbosdo committed:
Add virt.pool_get_xml function

Users may want to see the full XML definition of a virtual storage pool.

s0undt3ch committed:
Less salt function calls

garethgreenaway committed:
Porting #50261 to master.

garethgreenaway committed:
Porting #50261 to master.

garethgreenaway committed:
Porting #50261 to master.

dwoz committed:
Linter fixes

dwoz committed:
Linter fixes

dwoz committed:
Linter fixes

Akm0d committed:
Sort dict keys for consistent return

github-abcde committed:
Merge branch 'master' into mine-minion-acl-master

dwoz committed:
Merge pull request #55795 from twangboy/fix_win_lgpo

Fix issue with whitespace in ADML data

dwoz committed:
Merge pull request #55766 from waynew/update-contributing-docs

Update contributing docs

dwoz committed:
Merge pull request #55626 from waynew/stalebot-updates

Enhance Stalebot

dwoz committed:
Update yumpkg.py

Akm0d committed:
Merge branch 'master' into tenentcloud_201921

dwoz committed:
Fix merge wart

dwoz committed:
Merge pull request #55762 from cbosdo/virt-pool-net-xml

Add virt.(pool|network)_get_xml functions

dwoz committed:
Merge pull request #54236 from stratusjerry/winpkg_docs

Update windows-package-manager.rst

Ch3LL committed:
Add RAET deprecation to neon release notes

Ch3LL committed:
Merge remote-tracking branch 'upstream/master' into issue_49427

waynew committed:
Merge branch 'master' into master-yumpkg-do-not-report-uninstalled-patches-as-installed

Ch3LL committed:
add nacl import in try/except

lukasraska committed:
Always call salt.utils.crypt.reinit_crypto when forking during new job processing

cmcmarrow committed:
Merge branch 'master' into jboss_fix_2

dwoz committed:
Fix docs

dwoz committed:
Merge pull request #55805 from Ch3LL/raet_rn

Add RAET deprecation to neon release notes

dwoz committed:
Merge branch 'master' into master-prevent-crashing-if-ss-returns-ipv6-connections

Akm0d committed:
Merge branch 'master' into tenentcloud_201921

Akm0d committed:
Merge branch 'master' into master-proxmox-vm

brejoc committed:
Fix linting errors

brejoc committed:
Merge branch 'master' into master-mod-repo-enabled

dwoz committed:
Merge pull request #55635 from lukasraska/fix-crypto-sign

Always try to reinitialize Crypto.Random when forking minion process

dwoz committed:
Merge pull request #55760 from ogd-software/mine-minion-acl-master

Mine minion acl master

dwoz committed:
Merge pull request #55694 from garethgreenaway/55662_s3_pillar_fixes

[master] S3 Pillar Pagination Fixes

twangboy committed:
Bring #51372 to Master branch

twangboy committed:
Fix some lint

twangboy committed:
Fix an issue causing linux failures

s0undt3ch committed:
Increate allowed build time for Windows Builds under Py3

swiftgist committed:
Check retcode for tuned-adm and return

Signed-off-by: Eric Jackson <swiftgist@gmail.com>

twangboy committed:
Fix issue with overly long names

Some policy names were entire sentences pull from the adml file

s0undt3ch committed:
Amazon 1 also seems to suffer from the slowdown

twangboy committed:
Remove Microsoft Smart Characters with utf-8

Ch3LL committed:
Merge branch 'master' into develop

Ch3LL committed:
Merge remote-tracking branch 'upstream/master' into issue_49427

dwoz committed:
Merge pull request #55819 from s0undt3ch/features/pytest-master-6

Increate allowed build time for Windows Builds under Py3

dwoz committed:
Merge pull request #55707 from Akm0d/requires_salt_state_modules

Create a wrapper that checks for the presence of state modules for tests

Commit:
Fix Typos

dwoz committed:
Merge pull request #55751 from asomers/osfullname_freebsd

Set the osfullname grain on FreeBSD

dwoz committed:
Merge pull request #55616 from cmcmarrow/jboss_fix_2

jboss fix

dwoz committed:
Merge pull request #55060 from Akm0d/master-proxmox-vm

 Modify the proxmox VM when using clone in the salt-cloud profile if settings are present

twangboy committed:
Support old element naming

dwoz committed:
Merge pull request #54526 from Akm0d/tenentcloud_201921

Tencent Cloud 201921

twangboy committed:
Fix some lint

Add back commented code for later testing if there is an issue

twangboy committed:
Change warn_until to Phosphorus

dwoz committed:
Merge branch 'master' into 2019_2_1_port_51988

dwoz committed:
Merge branch 'master' into 2019_2_1_port_52352

dwoz committed:
Merge pull request #54639 from garethgreenaway/2019_2_1_port_50663

[master] Porting #50663 to master

twangboy committed:
Show Deprecation warning in comments

Add assert for comment field
Enable Phosphorus

meaksh committed:
Strip possible [] on IPv6 addresses returned by ss execution

srg91 committed:
Fix load cached grain "osrelease_info"

Before fix the module `loader._load_cached_grains` loaded `osrelease_info` grain as list.
But everythere it expects `tuple`.
Now we force `osrelease_info` type and it works as expected.

Fixes #54908

meaksh committed:
Exclude test_mod_del_repo_multiline_values for SUSE distros

meaksh committed:
Merge branch 'master' into master-use-checksums-args-in-postgres.datadir_init

meaksh committed:
Merge branch 'master' into master-fix-list_pkg-output-when-using-attr

twangboy committed:
Fix flakey test

dwoz committed:
Merge pull request #55545 from Ch3LL/issue_49427

Document deprecated keyfile and key args in nacl utils

dwoz committed:
Merge pull request #54664 from garethgreenaway/2019_2_1_port_51988

[master] Porting #51988 to master

twangboy committed:
Fix comments, add tests

dwoz committed:
Fix docs build

dwoz committed:
Merge pull request #55800 from s0undt3ch/features/pytest-master-3

Fix test base class + Reduce salt calls for system tests

dwoz committed:
Merge pull request #55578 from meaksh/master-use-checksums-args-in-postgres.datadir_init

[master] Use "checksums" args when calling "postgres.datadir_init"

twangboy committed:
Fix some issues, standardize extraction, fix names

twangboy committed:
Fix reg tests

twangboy committed:
Fix lint

dwoz committed:
Merge pull request #55582 from meaksh/master-yumpkg-do-not-report-uninstalled-patches-as-installed

[master] Do not report patches as installed when not all the related pkgs are installed (yumpkg)

dwoz committed:
Merge pull request #54640 from garethgreenaway/2019_2_1_port_52352

[master] Porting #52352 to master

dwoz committed:
Merge pull request #51844 from mbunkus/fix-file-removal

Fix file removal

s0undt3ch committed:
If the queue is set, return it.

dwoz committed:
Merge pull request #55830 from stratusjerry/TypoFix

Fix Typos

dwoz committed:
Skip lgpo unit tests on non windows platforms

dwoz committed:
Merge branch 'master' into master-fix-list_pkg-output-when-using-attr

dwoz committed:
Merge branch 'master' into requirements_mock

dwoz committed:
Fix linter

mchugh19 committed:
port #52212 to master

garethgreenaway committed:
Fixing failing tests.

garethgreenaway committed:
Fixing failing tests.

garethgreenaway committed:
Fixing failing tests.

garethgreenaway committed:
Fixing failing tests.

garethgreenaway committed:
Merge branch 'master_port_50261' of github.com:garethgreenaway/salt into master_port_50261

garethgreenaway committed:
Merge branch 'master' into master_port_50261

garethgreenaway committed:
Fixing lint, removing NO_MOCK and NO_MOCK_REASON.

twangboy committed:
Remove commented code

garethgreenaway committed:
Fixing lint

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

s0undt3ch committed:
We can't use spot instances if the max time is above 6 hours

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

garethgreenaway committed:
Merge branch 'master_port_50261' of github.com:garethgreenaway/salt into master_port_50261

dwoz committed:
Merge pull request #55846 from s0undt3ch/features/pytest-master-1

We can't use spot instances if the max time is above 6 hours

dwoz committed:
Merge pull request #55843 from mchugh19/port-52212

port #52212 to master - suggested for Neon

dwoz committed:
Merge pull request #55792 from asomers/requirements_mock

[skip ci] add mock to the list of requirements for testing

dwoz committed:
Merge pull request #55583 from meaksh/master-prevent-crashing-if-ss-returns-ipv6-connections

[master] Do not make Salt CLI to crash when there are IPv6 established connections

dwoz committed:
Merge branch 'master' into lgpo_skip

meaksh committed:
Fix pylint issues

dwoz committed:
Merge pull request #55844 from garethgreenaway/master_port_50059

[master] Porting #50059 to master

dwoz committed:
Merge pull request #55841 from dwoz/lgpo_skip

Skip lgpo unit tests on non windows platforms

dwoz committed:
Merge pull request #55796 from srg91/fix-cached-osrelease_info-grain-type

Fix cached osrelease_info grain type

brejoc committed:
Removal of blank line for linter

waynew committed:
Merge branch 'master' into fix-tuned-54010

s0undt3ch committed:
Merge branch 'master' into pass-grains-in-minion-startup-event

Ch3LL committed:
Merge branch 'master' into develop

dwoz committed:
Merge pull request #55845 from s0undt3ch/features/pytest-master-6

If the queue is set, return it.

dwoz committed:
Merge pull request #55203 from brejoc/master-mod-repo-enabled

Adds enabled kwarg

Akm0d committed:
Refactor the mac group tests

 - Simplify setup and teardown logic with user/group.present/absent
 - Added a function for asserting salt-call with no return code
 - Skip tests if required modules and states aren't available

dwoz committed:
Merge pull request #54948 from admd/pass-grains-in-minion-startup-event

Pass grains in minion startup event

meaksh committed:
Merge branch 'master' into master-yumpkg-do-not-break-repo-files-with-multiline-values

kiemlicz committed:
https://github.com/saltstack/salt/issues/51123

Removing usage of fetch unused port

Ch3LL committed:
Add test framework for pdbedit and state smartos tests

twangboy committed:
Increase timeout for test_long_running_job test

Commit:
Update release version

dwoz committed:
Merge pull request #55863 from frogunder/doc_conf_master

Update release version

dwoz committed:
Merge pull request #55580 from meaksh/master-fix-list_pkg-output-when-using-attr

[master] Fix inconsistency with `pkg.list_pkgs` when using `attr` on RHEL systems

dwoz committed:
Merge pull request #55584 from meaksh/master-yumpkg-do-not-break-repo-files-with-multiline-values

[master] Do not break repo files with multiple line values on "yumpkg"

kiemlicz committed:
Revert "https://github.com/saltstack/salt/issues/51123"

This reverts commit 093b735f7ae18f8b0d0dac0453089881006147a9.

dwoz committed:
Merge pull request #54072 from twangboy/fix_53955_develop

Check for key before trying to list it (master)

twangboy committed:
Fix enum settings

Ch3LL committed:
correctly mock pdbedit.list

garethgreenaway committed:
various netapi fixes and tests

garethgreenaway committed:
various netapi fixes and tests

garethgreenaway committed:
Removing version gating for installing mock 3.0.5

Ch3LL committed:
Merge branch 'master' into develop

Ch3LL committed:
Update release candidate docs to be more dynamic

Commit:
fix doc spaces

Ch3LL committed:
fix rc docs comma

Commit:
fix doc bug

Commit:
update doc

cmcmarrow committed:
Merge branch 'master' into branch_doc_update

Commit:
update neon reference

dwoz committed:
Merge pull request #55871 from Ch3LL/rc_docs

Update release candidate docs to be more dynamic

sagetherage committed:
More updates to stale.yml

adding more labels to stale bot config bug, feature, test failure

Commit:
Update 2016.11.9.rst

Commit:
Update 2016.11.9.rst

Commit:
Update 2017.7.3.rst

Commit:
Update 2018.3.1.rst

Commit:
Update 2018.3.3.rst

Commit:
Update bootstrap-salt.sh

Commit:
Update core.py

Commit:
Update aptpkg.py

Commit:
Update bootstrap-salt.sh

Commit:
Merge branch 'master' into update_neon_reference

Commit:
Update index.rst

cmcmarrow committed:
Merge branch 'master' into branch_doc_update

Commit:
Update man pages 3000

dwoz committed:
Merge pull request #55823 from twangboy/fix_lgpo_long_names

Fix issue with overly long names in the LGPO module

admd committed:
Restrict the 'start_event_grains' only to the start events

dwoz committed:
Merge pull request #55872 from cmcmarrow/add_space

fix doc spaces

dwoz committed:
Merge pull request #55610 from cmcmarrow/branch_doc_update

Update Branch Topology Doc

admd committed:
add test for custom events

garethgreenaway committed:
fixes to allow test_ssh to work.  including roster file options and options by pass the host key check.

garethgreenaway committed:
fixes to allow test_ssh to work.  including roster file options and options by pass the host key check.

garethgreenaway committed:
Merge remote-tracking branch 'upstream/master'

garethgreenaway committed:
Merge branch 'master_salt_netapi_fixes' of github.com:garethgreenaway/salt into master_salt_netapi_fixes

twangboy committed:
Fix issue with removed kwarg

dwoz committed:
Merge pull request #55861 from Ch3LL/smartos_test_templates

Add test framework for pdbedit and state smartos tests

dwoz committed:
Merge pull request #52980 from kiemlicz/develop

logstash_mod makePickle doesn't convert to bytes

dwoz committed:
Merge branch 'master' into fix-tuned-54010

dwoz committed:
Merge pull request #55821 from swiftgist/fix-tuned-54010

Fix tuned 54010

garethgreenaway committed:
Merge branch 'master' into master_port_50261

Commit:
the note only applies to SALT and SALT_SSH commands

Commit:
more neon/3000 updates

Commit:
Merge branch 'master' of https://github.com/saltstack/salt into update_neon_reference

Commit:
Merge branch 'master' into update_neon_reference

Commit:
Merge branch 'update_neon_reference' of https://github.com/frogunder/salt into update_neon_reference

Commit:
one more neon/3000 update

twangboy committed:
Add tests for the reg module

twangboy committed:
Fix some lint

twangboy committed:
Fix version-added, can't use 3000

dwoz committed:
Merge pull request #55807 from Akm0d/mac_group_tests

Mac group tests

twangboy committed:
Change version-added to versionadded

garethgreenaway committed:
Removing logging import to test if this resolves the test failure.

garethgreenaway committed:
Removing logging import to test if this resolves the test failure.

garethgreenaway committed:
Merge branch 'master_port_50261' of github.com:garethgreenaway/salt into master_port_50261

garethgreenaway committed:
Importing salt.log.setup so hopefully we have access to log.trace.

garethgreenaway committed:
Importing salt.log.setup so hopefully we have access to log.trace.

dwoz committed:
Merge pull request #55865 from garethgreenaway/master_salt_netapi_fixes

[master] various netapi fixes and tests

Commit:
Update version.py

Commit:
set release versions with environment variables in the docs

dwoz committed:
Merge pull request #55880 from frogunder/man_pages_3000

Update man pages 3000

dwoz committed:
Merge pull request #55875 from frogunder/update_neon_reference

update neon reference

Commit:
add previous release notes

garethgreenaway committed:
Adding some code to make sure log.trace is available.

garethgreenaway committed:
Adding some code to make sure log.trace is available.

dwoz committed:
Merge pull request #55903 from bryceml/make_docs_build_using_environment_variables_for_releases

set release versions with environment variables in the docs

twangboy committed:
Fix conflicts

twangboy committed:
Merge branch 'master' into fix_win_reg

twangboy committed:
Neon to 3000

twangboy committed:
Merge branch 'fix_win_reg' of https://github.com/twangboy/salt into fix_win_reg

dwoz committed:
Merge pull request #55893 from twangboy/fix_win_reg

Remove references to `include_default`

garethgreenaway committed:
Renaming beacon/log.py to beacon/log_beacon.py to avoid conflicts when importing logging during tests.

garethgreenaway committed:
Renaming beacon/log.py to beacon/log_beacon.py to avoid conflicts when importing logging during tests.

garethgreenaway committed:
Removing unneeded testing code.

garethgreenaway committed:
Removing unneeded testing code.

garethgreenaway committed:
Porting #50261 to master.

garethgreenaway committed:
Merge branch 'master_port_50261' of github.com:garethgreenaway/salt into master_port_50261

garethgreenaway committed:
Renaming test_log.py to test_log_beacon.py to fix unit.test_module_names.BadTestModuleNamesTestCase.test_module_name_source_match failure.

dwoz committed:
Initial commit of vendored tornado 4.5.3

Signed-off-by: Daniel A. Wozniak <dwozniak@saltstack.com>

dwoz committed:
Fix imports in vendored tornado

dwoz committed:
Add pylint skips to vendored tornado

dwoz committed:
Merge pull request #55885 from admd/startup_grains-only-in-start-event

Restrict the 'start_event_grains' only to the start events

dwoz committed:
Merge pull request #55794 from garethgreenaway/master_port_50261

[master] Porting #50261 to master - beacons

dwoz committed:
Merge pull request #55866 from garethgreenaway/update_mock_requirements

[master] Removing version gating for installing mock 3.0.5

dwoz committed:
Merge pull request #55858 from s0undt3ch/hotfix/venafy

Update Venfi Module

sagetherage committed:
Merge branch 'master' into sagetherage-stalebot2

dwoz committed:
Fix unit.test_doc

dwoz committed:
Merge pull request #55877 from saltstack/sagetherage-stalebot2

More updates to stale.yml

dwoz committed:
Merge pull request #55897 from vernondcole/doc_fix_cli_docs

the note only applies to SALT and SALT_SSH commands

dwoz committed:
Merge pull request #55817 from twangboy/win_network

Bring #51372 to Master branch

s0undt3ch committed:
Include all requirements files in source distribution

dwoz committed:
Merge pull request #55927 from dwoz/vendored_tornado

Add tornado 4.5.3 codebase to salt/ext

dwoz committed:
Merge pull request #55950 from s0undt3ch/hotfix/requirements

Include all requirements files in source distribution

dwoz committed:
Fix linter failures that came up recently

dwoz committed:
Merge pull request #55965 from dwoz/linter-fixes

Fix linter failures that came up recently

garethgreenaway committed:
Removing NO_MOCK and NO_MOCK_REASON

garethgreenaway committed:
Removing unused imports

garethgreenaway committed:
Removing test_start.  Lint cleanup.

garethgreenaway committed:
Merge branch '55057_stalekey_engine_fixes' of github.com:garethgreenaway/salt into 55057_stalekey_engine_fixes
